src/main/java/com/exadel/discount/service/impl/CountryServiceImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.exadel.discount.service.impl;
+
+import com.exadel.discount.dto.CountryDTO;
+import com.exadel.discount.exception.NotFoundException;
+import com.exadel.discount.mapper.CountryMapper;
+import com.exadel.discount.repository.CountryRepository;
+import com.exadel.discount.service.CountryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@RequiredArgsConstructor
+public class CountryServiceImpl implements CountryService {
+    private final CountryRepository countryRepository;
+    private final CountryMapper countryMapper;
+
+    @Override
+    public List<CountryDTO> findAll() {
+        return countryMapper.getListDTO(countryRepository.findAll());
+    }
+
+    @Override
+    public CountryDTO findOneById(UUID id) {
+
+        return countryMapper.getDTO(countryRepository.findById(id)
+                .orElseThrow(() -> new NotFoundException(String.format("Country with id %s not found", id))));
+    }
+
+    @Override
+    public CountryDTO findByName(String name) {
+
+        return countryMapper.getDTO(countryRepository.findAll()
```

Comment:
Are U going to add Slf4j in Service layer? 
So It willl be changed into more row code.
Like here.
https://github.com/SergeyKoval/2021_june_team4_be/blob/dev/src/main/java/com/exadel/discount/service/impl/TagServiceImpl.java

Reply:
Done.

Reply:
> 
> 
> Are U going to add Slf4j in Service layer?
> So It willl be changed into more row code.
> Like here.
> https://github.com/SergeyKoval/2021_june_team4_be/blob/dev/src/main/java/com/exadel/discount/service/impl/TagServiceImpl.java

Done.